hs_modflow_modelinstance/models.py

```diff
@@ -0,0 +1,799 @@
+                                               ('DRT', 'DRT'), ('ETS', 'ETS'), ('EVT', 'EVT'),
+                                               ('GHB', 'GHB'), ('LAK', 'LAK'), ('MNW1', 'MNW1'),
+                                               ('MNW2', 'MNW2'), ('RES', 'RES'), ('RIP', 'RIP'),
+                                               ('RIV', 'RIV'), ('SFR', 'SFR'), ('STR', 'STR'),
+                                               ('UZF', 'UZF'),)
+    specified_head_boundary_packages = models.ManyToManyField(SpecifiedHeadBoundaryPackageChoices,
+                                                              blank=True)
+    other_specified_head_boundary_packages = models.CharField(max_length=200, null=True, blank=True,
+                                                              verbose_name='Other packages')
+    specified_flux_boundary_packages = models.ManyToManyField(SpecifiedFluxBoundaryPackageChoices,
+                                                              blank=True)
+    other_specified_flux_boundary_packages = models.CharField(max_length=200, null=True, blank=True,
+                                                              verbose_name='Other packages')
+    head_dependent_flux_boundary_packages = \
+        models.ManyToManyField(HeadDependentFluxBoundaryPackageChoices, blank=True)
+    other_head_dependent_flux_boundary_packages = models.CharField(max_length=200, null=True,
+                                                                   blank=True,
+                                                                   verbose_name='Other packages')
+
+    # may be this could cause a problem
+    def __unicode__(self):
+        return self.term
+
+    class Meta:
+        # BoundaryCondition element is not repeatable
+        unique_together = ("content_type", "object_id")
+
+    def get_specified_head_boundary_packages(self):
+        return ', '.join([types.description for types in
+                          self.specified_head_boundary_packages.all()])
+
+    def get_specified_flux_boundary_packages(self):
+        return ', '.join([packages.description for packages in
+                          self.specified_flux_boundary_packages.all()])
+
+    def get_head_dependent_flux_boundary_packages(self):
+        return ', '.join([packages.description for packages in
+                          self.head_dependent_flux_boundary_packages.all()])
+
+    @classmethod
+    def _add_specified_head_boundary_packages(cls, boundary_packages, packages):
+        for package in packages:
+            qs = SpecifiedHeadBoundaryPackageChoices.objects.filter(description__exact=package)
+            if qs.exists():
+                boundary_packages.specified_head_boundary_packages.add(qs[0])
+            else:
+                boundary_packages.specified_head_boundary_packages.create(description=package)
+
+    @classmethod
+    def _add_specified_flux_boundary_packages(cls, boundary_packages, packages):

```

Comment:
These ```_add ....._packages()``` functions be better if there are instance methods and not class methods. Then it  can be used as:

```model_boundary_condition._add_specified_head_boundary_packages( kwargs['specified_head_boundary_packages'])```